lib/chromic_pdf/supervisor.ex

```diff
@@ -26,69 +26,38 @@ defmodule ChromicPDF.Supervisor do
       end
   """
 
-  # disables the @doc for child_spec/1
-  @doc false
-  use Supervisor
-  import ChromicPDF.Utils, only: [find_supervisor_child: 2]
+  import ChromicPDF.Utils, only: [find_supervisor_child: 2, find_supervisor_child!: 2]
   alias ChromicPDF.{Browser, GhostscriptPool}
 
   @type services :: %{
           browser: pid(),
           ghostscript_pool: pid()
         }
 
-  defp on_demand?(config), do: Keyword.get(config, :on_demand, false)
-  defp on_demand_name(chromic_name), do: Module.concat(chromic_name, :OnDemand)
-
-  @doc """
-  Returns a specification to start this module as part of a supervision tree.
-  """
   @spec child_spec([ChromicPDF.global_option()]) :: Supervisor.child_spec()
-  def child_spec(chromic, config) do
-    type =
-      if on_demand?(config) do
-        :worker
-      else
-        :supervisor
-      end
-
+  def child_spec(config) do
     %{
-      id: chromic,
-      start: {chromic, :start_link, [config]},
-      type: type
+      id: Keyword.fetch!(config, :name),
```

Comment:
drive-by: also make the `id` dependent on the name to avoid conflicts when dynamic names are used